llmfoundry/utils/exceptions.py

```diff
@@ -354,3 +356,26 @@ class RunTimeoutError(InternalError):
     def __init__(self, timeout: int) -> None:
         message = f'Run timed out after {timeout} seconds.'
         super().__init__(message, timeout=timeout)
+
+
+class DatasetMissingFileError(UserError):
+    """Error thrown when a dataset cannot find a file."""
+
+    def __init__(
+        self,
+        file_name: str,
+        supported_extensions: Any,  # Run into type error when using List[str]
+    ) -> None:
+        message = "Could not find the file '{file_name}' with any of the supported extensions: "
+        message += ', '.join(supported_extensions) + '.'
+        message += ' Please check your train / eval data and try again.'
```

Comment:
```suggestion
        message += ' Please check your train / eval dataset configuration and try again.'
```
Not sure how this will be surfaced or if users will have access to this info, but this is more specific at least